File: packages/app/internal/services/app_info.go
package services

import (
	"github.com/chenyang-zz/cove/internal/domain"
	"github.com/chenyang-zz/cove/internal/platform"
)

type AppInfoService struct {
	name    string
	version string
}

func NewAppInfoService(name, version string) *AppInfoService {
	return &AppInfoService{
		name:    name,
		version: version,
	}
}

func (s *AppInfoService) GetAppInfo() (domain.AppInfo, error) {
	system := platform.CurrentSystemInfo()

	return domain.AppInfo{
		Name:     s.name,
		Version:  s.version,
		Platform: system.Platform,
		Arch:     system.Arch,
	}, nil
}

File: packages/app/frontend/src/shared/api/appInfo.ts
import { AppInfoService } from '../../../bindings/github.com/chenyang-zz/cove/internal/services'

export type AppInfo = {
  name: string
  version: string
  platform: string
  arch: string
}

export async function getAppInfo(): Promise<AppInfo> {
  return AppInfoService.GetAppInfo()
}
